ext/standard/math.c

```diff
@@ -1328,6 +1328,16 @@ PHP_FUNCTION(number_format)
 			break;
 
 		case IS_DOUBLE:
+			// double values of >= 2^52 can not have fractional digits anymore
+			// Casting to long on 64bit will not loose precision on rounding
+			if (UNEXPECTED(
+				(Z_DVAL_P(num) >= 4503599627370496.0 || Z_DVAL_P(num) <= -4503599627370496.0)
+				&& ZEND_DOUBLE_FITS_LONG(Z_DVAL_P(num))
```

Comment:
Isn't `ZEND_DOUBLE_FITS_LONG(Z_DVAL_P(num))` a little bit faster than the first condition? We have the difference of one execution of `Z_DVAL_P`.

Reply:
I don't think so.
1. This is first checking the unusual case for numbers above 2^52
2. `ZEND_DOUBLE_FITS_LONG` will cast `ZEND_LONG_[MIN|MAX]` and compare double values as well `!((d) >= (double)ZEND_LONG_MAX || (d) < (double)ZEND_LONG_MIN)` 
3. `Z_DVAL_P(num)`  results in `*num.value.dval`

Reply:
For number outside the order doesn't matter because two conditions must be met. But for number inside the range the second condition has only one `Z_DVAL_P` evaluation and first condition has two of them. The comparison of doubles occurs in both conditions.

However, I think that the difference is so small that you can ignore my comment. Probably even it may be optimized by compiler.